src/importlinter/adapters/user_options.py

```diff
@@ -51,3 +61,38 @@ def _clean_section_config(section_config: Dict[str, Any]) -> Dict[str, Any]:
             else:
                 section_dict[key] = value.strip().split("\n")
         return section_dict
+
+
+class TomlUserOptionReader(ports.UserOptionReader):
+    """
+    Reader that looks for and parses the contents of TOML files.
+    """
+
+    section_name = "importlinter"
+
+    def read_options(self, config_filename: Optional[str] = None) -> Optional[UserOptions]:
+        if not _HAS_TOML:
+            return None
+
+        if config_filename:
+            config_filenames = file_finding.find_any(config_filename)
+            if not config_filenames:
+                raise FileNotFoundError(f"Could not find {config_filename}.")
+        else:
+            config_filenames = file_finding.find_any("pyproject.toml")
```

Comment:
There are a lot of similarities in the config filenames logic across both classes. Can we extract the common behaviour into a base class?

In particular, I think it's worth using `potential_config_filenames` as an attribute on both classes so we can just iterate through them - rather than hardcoding `pyproject.toml` in the method body as it is here.

Reply:
done